.github/workflows/smoke-test.yml

```
@@ -0,0 +1,215 @@
+        include:
+          - platform: ubuntu-latest
+            os: Linux
+          - platform: macos-latest
+            os: macOS
+          - platform: windows-latest
+            os: Windows
+
+    runs-on: ${{ matrix.platform }}
+
+    # Skip macOS on PRs to conserve GH Actions minutes (10x multiplier)
+    if: ${{ matrix.os != 'macOS' || github.event_name == 'push' }}
```

actionlint (1.7.12)

[error] context "matrix" is not allowed here. available contexts are "github", "inputs", "needs", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)